packages/rainbowkit/src/components/WalletButton/WalletButtonRenderer.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { ReactNode, useContext, useEffect, useState } from 'react';
+import { BaseError } from 'viem';
+import { useAccount } from 'wagmi';
+import { useConnectionStatus } from '../../hooks/useConnectionStatus';
+import { useIsMounted } from '../../hooks/useIsMounted';
+import { isMobile } from '../../utils/isMobile';
+import {
+  WalletConnector,
+  useWalletConnectors,
+} from '../../wallets/useWalletConnectors';
+import {
+  useConnectModal,
+  useModalState,
+} from '../RainbowKitProvider/ModalContext';
+import { RainbowButtonContext } from '../RainbowKitProvider/RainbowButtonContext';
+
+export interface WalletButtonRendererProps {
+  children: (renderProps: {
+    ready: boolean;
+    connect: () => void;
+    loading: boolean;
+    error: string;
+    connector: WalletConnector;
+  }) => ReactNode;
+}
+
+export function WalletButtonRenderer({ children }: WalletButtonRendererProps) {
+  const mounted = useIsMounted();
+  const { openConnectModal } = useConnectModal();
+  const { connectModalOpen } = useModalState();
+  const { connector, setConnector } = useContext(RainbowButtonContext);
+  const [firstConnector] = useWalletConnectors()
+    .filter((wallet) => wallet.ready)
+    .sort((a, b) => a.groupIndex - b.groupIndex);
```

Comment:
@DanielSinclair, for `WalletButton` API connector sharing logic, when given connectors like `connectors: [metamaskWallet()]`, we'll always use the first one. This is the same for connectors: `[new RainbowConnector()]`. We can't pass only one connector to the API. 

If developers don't want the first choice, they might need a specific `connector` prop which they can pass in connector themselves. But generally, this works fine. Let me know what you think.

Reply:
We'll have to work through this one. If the user adopts the `WalletButton` component, they'll need to pass through a connector id like `<WalletButton wallet='metamask'/>` and we'll need to select the connector through that path. The user could use multiple different WalletButtons simultaneously

Reply:
We also can't control the order of the connectors list, as the dApp will use other connectors for their custom injected/walletconnector behavior or other wallet buttons

Reply:
Oh yeah i was thinking the same. Passing some sort of prop to the component. I thought we were initally going to pass what's in the `connectors` from wagmi initially.